lib/rules/prefer-explicit-assert.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const { getDocsUrl, ALL_QUERIES_METHODS } = require('../utils');
+
+const ALL_GET_BY_QUERIES = ALL_QUERIES_METHODS.map(
+  queryMethod => `get${queryMethod}`
+);
+
+const isValidQuery = (node, customQueries = []) =>
+  ALL_GET_BY_QUERIES.includes(node.name) || customQueries.includes(node.name);
+
+const isDirectlyCalledByFunction = node =>
+  node.parent.parent.type === 'CallExpression';
+
+const isReturnedByArrowFunctionExpression = node =>
+  node.parent.parent.type === 'ArrowFunctionExpression';
+
+const isDeclared = node => node.parent.parent.type === 'VariableDeclarator';
+
+const isReturnedByReturnStatement = node =>
+  node.parent.parent.type === 'ReturnStatement';
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Suggest using explicit assertions rather than just `getBy*` queries',
+      category: 'Best Practices',
+      recommended: false,
+      url: getDocsUrl('prefer-explicit-assert'),
+    },
+    messages: {
+      noGetByAssert:
```

Comment:
For consistency, I think it's better to rename the message id too: `preferExplicitAssert`

Reply:
Oops, good catch! Let me fix that.

Reply:
Done, I need your approval again 😛

Reply:
Just saw @emmenko comment on another test case so I'll wait for it to be included before approving 😉